disperser/controller/metadata/batch_metadata_manager.go

```diff
@@ -0,0 +1,199 @@
+package metadata
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"sync/atomic"
+	"time"
+
+	regcoordinator "github.com/Layr-Labs/eigenda/contracts/bindings/RegistryCoordinator"
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+)
+
+// An object responsible for acquiring and providing batch metadata (i.e. operator state and reference block number)
+// for the creation of new batches.
+type BatchMetadataManager interface {
+
+	// GetMetadata returns the metadata required to create a new batch. Although the data will be updated periodically,
+	// this utility makes no guarantees about the freshness of the data returned by this method. Keeping up to date
+	// with the most recent onchain data is done on a best effort basis.
+	GetMetadata() *BatchMetadata
+
+	// Release resources associated with this manager.
+	Close()
+}
+
+var _ BatchMetadataManager = (*batchMetadataManager)(nil)
+
+// A standard implementation of the BatchMetadataManager interface. Does all metadata fetching in a background\
+// goroutine, guaranteeing that GetMetadata() never blocks.
+type batchMetadataManager struct {
+	ctx    context.Context
+	logger logging.Logger
+
+	// The underlying eth client used to interact with the blockchain.
+	ethClient bind.ContractBackend
+
+	// Used to get operator state. The IndexedChainState utility fetches state both from onchain sources and from
+	// the indexer. When we eventually move all data onchain, we can ditch the indexer and just call directly
+	// into the contract bindings in this file.
+	indexedChainState core.IndexedChainState
+
+	// A handle for communicating with the registry coordinator contract.
+	registryCoordinator *regcoordinator.ContractRegistryCoordinator
+
+	// When choosing a new reference block number (RBN), select the block that is this many blocks in the past.
+	// This is a hedge against forking.
+	referenceBlockOffset uint64
+
+	// The time between updates to the metadata.
+	updatePeriod time.Duration
+
+	// The most recent batch metadata.
+	metadata atomic.Pointer[BatchMetadata]
+
+	alive atomic.Bool
+}
+
+// Create a new BatchMetadataManager with the specified quorums.
+//
+// updatePeriod is the period at which the manager will update its metadata. Actual update timing may vary
+// depending on the amount of time it takes to successfully get new data.
```

Comment:
A little more info in this constructor would be nice:

- synchronously does the first fetch
- starts the background process to periodically update